qiskit/circuit/controlledgate.py

```diff
@@ -169,6 +167,55 @@ def ctrl_state(self, ctrl_state: Union[int, str, None]):
             raise CircuitError('invalid control state specification: {}'.format(
                 repr(ctrl_state)))
 
+    @property
+    def params(self):
+        """Get parameters from base_gate.
+
+        Returns:
+            list: List of gate parameters.
+
+        Raises:
+            CircuitError: Controlled gate does not define a base gate
+        """
+        if self.base_gate:
+            return self.base_gate.params
+        else:
+            raise CircuitError('Controlled gate does not define base gate for '
+                               'for extracting params')
+
+    @params.setter
+    def params(self, parameters):
+        """Set base gate parameters.
+
+        Args:
+            parameters (list): The list of parameters to set.
+
+        Raises:
+            CircuitError: If controlled gate does not define a base gate.
+        """
+        if self.base_gate:
+            super(Gate, self.__class__).params.fset(self.base_gate, parameters)
+            if parameters:
+                new_params = []
+                for param in parameters:
+                    if isinstance(param, Parameter):
+                        new_params.append(copy.copy(param))
```

Comment:
Do these need to be copied? This prevents setting the parameters to concrete parameter instances I think. E.g.
```python
p = Parameter('p')
cgate.params = [p]
cgate.params[0] is p  # False?
```
which can be problematic if we want to append it to a circuit due to name conflicts and not being the same instance.

Reply:
Looks like I was able to remove the copy here. Thanks.